frontend/README.md

```diff
@@ -2,28 +2,77 @@
 
 A single-page app for the [OpenCraft Instance Manager (OCIM)](https://github.com/open-craft/opencraft).
 
-# Provisioning
+## Provisioning
 
-1. Install the API client:
-```
+- Install the API client:
+
+```bash
 ./scripts/build-api-client.sh
 ```
-2. Install requirements:
-```
+
+- Install requirements:
+
+```bash
 npm install
 ```
-3. Run frontend:
-```
+
+- Run frontend:
+
+```bash
 npm start
 ```
 
-# Running with Vagrant
+- Generating New Components:
+
+```bash
+npm run generate # uses plop, more information here(https://plopjs.com/)
+```
+
+## API Client Instructions
+
+- Updating the API Client:
+
+```bash
+cd ./frontend # go to frontend directory
+npm run update-api-client
+```
+
+- Building the API Client:
+
+```bash
+cd ./frontend # go to frontend directory
+npm run build-api-client
+```
+
+## Testing
+
+- Running the tests:
+
+```bash
+npm run lint # for running the linting
+
+npm run lint-fix # for fixing the linting
```

Comment:
This just runs eslint with the `--fix` flag so it tries to correct as many issues as it can, but it'll still fail in case it doesn't manage to fix something.

Reply:
Added the details in the comments to make it more clear